config/routes.rb

```diff
@@ -11,8 +12,19 @@
 
   get "/robots.txt" => RobotsGenerator
 
-  get "/sitemap.xml.gz"          => "sitemap#sitemap", format: :xml
-  get "/sitemap/generate.xml.gz" => "sitemap#generate", format: :xml
+  # URLs for sitemaps
+  #
+  # From Rails guide: By default dynamic segments don’t accept dots –
+  # this is because the dot is used as a separator for formatted
+  # routes. If you need to use a dot within a dynamic segment add a
+  # constraint which overrides this – for example :id => /[^\/]+/
+  # allows anything except a slash.
+  #
+  # That's why there's the constraints in generate URL to accept host
+  # parameter with dots
+  #
+  get "/sitemap.xml.gz"                        => "sitemap#sitemap", format: :xml
+  get "/sitemap/:sitemap_host/generate.xml.gz" => "sitemap#generate", format: :xml, :constraints => { :sitemap_host => /[^\/]+/ }
```

Comment:
How about making the regex a bit more tight? E.g. `/[.-\w]+/` (dash, alphanum and dot)?